src/Lecture3.hs

```diff
@@ -247,6 +279,12 @@ Nothing
 Just [8,9,10]
 >>> apply 5 [(+ 3), (* 4), div 17]
 [8,20,3]
-
 -}
+
 apply = error "TODO"
+{- 
+apply :: Functor f => a -> f (a -> b) -> f b
+apply _ mempty = mempty
+apply z (f x) = f (fmap x z)
```

Comment:
The type signature of `apply` function isi correct 💯 

Think in the following way: you have a functor value of type `f (a -> b)` and you need to get `f b`. The only thing you can do with the `Functor` is to call `fmap`. So your implementation will be something like this:

```haskell
apply x functor = fmap someFunctionHere functor
```

So the challenge is in implementing this `someFunctionHere`. Try to think about its type 🙂 
And if it's still not clear yet, you can use a Haskell feature called "Typed holes" to ask the compiler for the type of this function. So just write:

```
apply x functor = fmap _fun functor
```

And GHC will tell you the expected type of `_fun`

I hope this will help to move with this task 🙂